domain/account.go

```diff
@@ -106,54 +90,37 @@ func openAccount(s dogma.AggregateCommandScope, m commands.OpenAccount) {
 	})
 }
 
-func creditForDeposit(s dogma.AggregateCommandScope, m commands.CreditAccountForDeposit) {
-	s.RecordEvent(events.AccountCreditedForDeposit{
-		TransactionID: m.TransactionID,
-		AccountID:     m.AccountID,
-		Amount:        m.Amount,
-	})
-}
-
-func creditForTransfer(s dogma.AggregateCommandScope, m commands.CreditAccountForTransfer) {
-	s.RecordEvent(events.AccountCreditedForTransfer{
-		TransactionID: m.TransactionID,
-		AccountID:     m.AccountID,
-		Amount:        m.Amount,
+func creditAccount(s dogma.AggregateCommandScope, m commands.CreditAccount) {
+	s.RecordEvent(events.AccountCredited{
+		TransactionID:   m.TransactionID,
+		AccountID:       m.AccountID,
+		TransactionType: m.TransactionType,
+		Amount:          m.Amount,
 	})
 }
 
-func debitForWithdrawal(s dogma.AggregateCommandScope, m commands.DebitAccountForWithdrawal) {
+func debitAccount(s dogma.AggregateCommandScope, m commands.DebitAccount) {
 	r := s.Root().(*account)
 
-	if r.Balance >= m.Amount {
-		s.RecordEvent(events.AccountDebitedForWithdrawal{
-			TransactionID: m.TransactionID,
-			AccountID:     m.AccountID,
-			Amount:        m.Amount,
+	if r.hasAvailableAmount(m.Amount) {
+		s.RecordEvent(events.AccountDebited{
+			TransactionID:   m.TransactionID,
+			AccountID:       m.AccountID,
+			TransactionType: m.TransactionType,
+			Amount:          m.Amount,
+			ScheduledDate:   m.ScheduledDate,
 		})
 	} else {
-		s.RecordEvent(events.WithdrawalDeclinedDueToInsufficientFunds{
-			TransactionID: m.TransactionID,
-			AccountID:     m.AccountID,
-			Amount:        m.Amount,
+		s.RecordEvent(events.AccountDebitDeclined{
+			TransactionID:   m.TransactionID,
+			AccountID:       m.AccountID,
+			TransactionType: m.TransactionType,
+			Amount:          m.Amount,
+			Reason:          messages.InsufficientFunds,
 		})
 	}
 }
 
-func debitForTransfer(s dogma.AggregateCommandScope, m commands.DebitAccountForTransfer) {
-	r := s.Root().(*account)
-
-	if r.Balance >= m.Amount {
-		s.RecordEvent(events.AccountDebitedForTransfer{
-			TransactionID: m.TransactionID,
-			AccountID:     m.AccountID,
-			Amount:        m.Amount,
-		})
-	} else {
-		s.RecordEvent(events.TransferDeclinedDueToInsufficientFunds{
-			TransactionID: m.TransactionID,
-			AccountID:     m.AccountID,
-			Amount:        m.Amount,
-		})
-	}
+func (r *account) hasAvailableAmount(amount int64) bool {
```

Comment:
I think the method does make it slightly clearer, but I'm happy to remove it if you think its not needed. For the example it kind of shows that its ok to add helper methods.

Reply:
Yeah, ok.  Perhaps it could be named inline with the ubiquitous language, something like `hasSufficientFunds()`.  The body can also be a simple comparison instead of a subtraction. I'm nit picking, but the whole point is to show how simple this can be, hopefully :)

Reply:
Ok, that makes it much simplier. If it wasn't for it now being in the ubiquitous language I would be more inclined to remove it.
```
func (r *account) hasSufficientFunds(amount int64) bool {
	return r.Balance >= amount
}
```